tools/syz-declextract/run.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/tool"
+)
+
+type compileCommand struct {
+	Arguments []string
+	Directory string
+	File      string
+	Output    string
+}
+
+func run(file string, out chan []byte) {
```

Comment:
Move this function after main.
Go (unlike C/C++) does not require things to be pre-declared, so it's not necessary to throw the most uninteresting implementation details onto reader first and force reading source file from bottom up.
In Go it's possible to declare more important things first, and less important things later.

Reply:
Done!